### PR TITLE
Tests for Lattice3D class

### PR DESCRIPTION
**Pull request by NGoetz**
This adds tests for the Lattice3D classes, and fixes some bugs, adds sanity checks and streamlines unknown values treatment for Particle class.

### CODE REVIEW

**Review by Hendrik1704 — APPROVED** [review]
Looks good from my side. I made some cosmetic changes in the way the ValueErrors are described in the documentation of two Lattice3D functions and I added the missing license header in the tests for Lattice3D.

**Comment by NGoetz** [comment]
@nilssass we wait for your approval as we both have committed this.

**Comment by nilssass** [comment]
This looks fine to me

**Review by nilssass — APPROVED** [review]

**Pull request merged**
